website/client/src/components/shops/balanceInfo.vue

```diff
@@ -67,33 +71,40 @@ export default {
   computed: {
     currencies () {
       const currencies = [];
-      currencies.push({
-        type: 'hourglasses',
-        icon: this.icons.hourglasses,
-        value: this.userHourglasses,
-      });
+      if (this.neededCurrencyOnly === true) {
+        currencies.push({
+          type: 'gems',
+          icon: this.icons.gem,
+          value: this.userGems,
+        });
+      } else {
+        currencies.push({
+          type: 'hourglasses',
+          icon: this.icons.hourglasses,
+          value: this.userHourglasses,
+        });
 
-      currencies.push({
-        type: 'gems',
-        icon: this.icons.gem,
-        value: this.userGems,
-      });
+        currencies.push({
+          type: 'gems',
+          icon: this.icons.gem,
+          value: this.userGems,
+        });
 
-      currencies.push({
-        type: 'gold',
-        icon: this.icons.gold,
-        value: this.userGold,
-      });
+        currencies.push({
+          type: 'gold',
+          icon: this.icons.gold,
+          value: this.userGold,
+        });
 
-      for (const currency of currencies) {
-        if (
-          currency.type === this.currencyNeeded
-          && !this.enoughCurrency(this.currencyNeeded, this.amountNeeded)
-        ) {
-          currency.notEnough = true;
+        for (const currency of currencies) {
```

Comment:
You're skipping this code block in the needed-only case. Don't we still want the component to notice when the user doesn't have enough of the needed currency, even if we're only showing one?

Reply:
Still thinking this one through.

Reply:
Still bugged per later comment; resolving this one for now